packages/server/graphql/intranetSchema/mutations/stripeUpdateInvoiceItem.ts

```diff
@@ -1,22 +1,76 @@
 import {GraphQLBoolean, GraphQLID, GraphQLNonNull} from 'graphql'
-import {InternalContext} from '../../graphql'
-import {isSuperUser} from '../../../utils/authorization'
+import Stripe from 'stripe'
 import getRethink from '../../../database/rethinkDriver'
-import StripeManager from '../../../utils/StripeManager'
 import InvoiceItemHook from '../../../database/types/InvoiceItemHook'
+import {isSuperUser} from '../../../utils/authorization'
+import sendToSentry from '../../../utils/sendToSentry'
+import StripeManager from '../../../utils/StripeManager'
+import {InternalContext} from '../../graphql'
 
 const MAX_STRIPE_DELAY = 3 // seconds
 
-const getBestHook = (possibleHooks: InvoiceItemHook[], hookQuantity: number) => {
+const getBestHook = (possibleHooks: InvoiceItemHook[]) => {
   if (possibleHooks.length === 1) return possibleHooks[0]
-  for (let i = 0; i < possibleHooks.length; i++) {
-    const curHook = possibleHooks[i]
-    const {quantity} = curHook
-    // if 2 hooks occurred at the same second, try grabbing the one with the accurate quantity
-    if (quantity === hookQuantity) return curHook
+  const firstHook = possibleHooks[possibleHooks.length - 1]
+  const {id: hookId} = firstHook
+  sendToSentry(new Error('Imperfect invoice item hook selected'), {tags: {hookId}})
+  return firstHook
+}
+
+const tagInvoiceItemWithHook = async (invoiceItem: Stripe.invoiceItems.InvoiceItem) => {
```

Comment:
@jordanh please review for atomicity. this webhook may get called by stripe multiple times per second